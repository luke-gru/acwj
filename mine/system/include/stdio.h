#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
