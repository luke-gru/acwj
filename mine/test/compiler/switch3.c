#include <stdio.h>

int main() {
  int x = 0;
  int y = 10;

  switch (x) {
    case 0:
    case 1:
    case 2:
      switch (y) {
        case 10:
          printf("breaking from inner with %d\n", y);
          break;
        case 20:
          printf("Error at 20\n");
        default:
          printf("Error!\n");
      }
      printf("non-default: %d\n", x);
      break;
    default:
      printf("default: %d\n", x);
      break;
  }
  printf("finished\n");
  return (0);
}
