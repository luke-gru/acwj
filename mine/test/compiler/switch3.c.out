breaking from inner with 10
non-default: 0
finished
